Prefix controller resource names with 'argocd-'

zcahana opened the pull request:
This PR renames the various controller resources (and their corresponding YAML files) such that they'll be prefixed with 'argocd-', similarly to other core Argo CD resources.

Aside from improving consistency, this clarifies the origin of non-namespaced Argo CD resources such as `argocd-application-controller-clusterrole` (previously named `application-controller-clusterrole`).

alexmt commented:
It makes sense to me. Thank you for contribution @zcahana !

@jessesuen, what do you think?

jessesuen commented:
Makes sense to me. Go for it

zcahana commented on manifests/base/argocd-application-controller-service.yaml:
```diff
@@ -1,10 +1,10 @@
 kind: Service
 metadata:
-  name: application-controller
+  name: argocd-application-controller
```
Great catch! Fixing...

mgoodness commented:
I applaud this change! May I also recommend dropping the resource type from some of the resource names?

*-role
*-role-binding
*-clusterrole
*-clusterrolebinding

zcahana commented:
Per @mgoodness' suggestion, I've also renamed the various `[Cluster]Role[Binding]` resources to omit the resource type from their names. @alexmt I hope you're good with this change too, otherwise I can revert.
Also rebased and resolved a minor conflict.

alexmt commented:
Looks good. Thanks for contribution @zcahana !
